OSDOCS-34823: updates migration procedure SDN to OVN-K (#78460)
openshift/openshift-docs: modules/nw-ovn-kubernetes-migration.adoc

$ oc get Network.config.openshift.io cluster -o yaml > cluster-openshift-sdn.yaml
----

. Verify that the  `OVN_SDN_MIGRATION_TIMEOUT` environment variable is set and is equal to `0s` by running the following command:
+
[source,bash]
----
#!/bin/bash

if [ -n "$OVN_SDN_MIGRATION_TIMEOUT" ] && [ "$OVN_SDN_MIGRATION_TIMEOUT" = "0s" ]; then
    unset OVN_SDN_MIGRATION_TIMEOUT

> fi

Reviewer: Should this be `if`?

Author: That looks right to me. It closes the if statement. I'm not an engr though so I'll double check.

#loops the timeout command of the script to repeatedly check the cluster Operators until all are available.

co_timeout=${OVN_SDN_MIGRATION_TIMEOUT:-1200s}
timeout "$co_timeout" bash <<EOT
until
  oc wait co --all --for='condition=AVAILABLE=True' --timeout=10s && \
  oc wait co --all --for='condition=PROGRESSING=False' --timeout=10s && \
  oc wait co --all --for='condition=DEGRADED=False' --timeout=10s;
do
  sleep 10
  echo "Some ClusterOperators Degraded=False,Progressing=True,or Available=False";
done
EOT
----

. Remove the configuration from the Cluster Network Operator (CNO) configuration object by running the following command:
+
[source,terminal]
----
oc patch Network.operator.openshift.io cluster --type='merge' /  --patch '{"spec":{"migration":null}}'
----

> . To prepare all the nodes for the migration, set the `migration` field on the Cluster Network Operator configuration object by entering the following command:

Reviewer: Because you introduced the CNO abbreviation in step 3, you can use it here.
```suggestion
. To prepare all the nodes for the migration, set the `migration` field on the CNO configuration object by running the following command:
```
I also changed "entering" to "running" for consistency, as that looks like the terminology mainly used in this procedure.

Author: I'll grab this while I'm here for sure. Thanks.

This step does not deploy OVN-Kubernetes immediately. Instead, specifying the `migration` field triggers the Machine Config Operator (MCO) to apply new machine configs to all the nodes in the cluster in preparation for the OVN-Kubernetes deployment.
====

.. Check that the reboot is finished by running the following command:
+
[source,terminal]
----
$ oc get mcp
----

.. Check that all cluster Operators are available by running the following command:
+
[source,terminal]
----
$ oc get co
----

> .. Alternatively: You can disable automatic migration of several OpenShift SDN capabilities to the OVN-Kubernetes equivalents:

Reviewer: Is this an optional step or an alternative step? If it's an alternative step, is it an alternative to steps a and b? If it's an optional step, replace "Alternatively" with "Optional", per IBM Style: https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#indicating-optional-and-conditional-steps

Author: Tricky question. It was an optional step that QE and SME have said is an `Alternative` step. That's actually a big part of this bug 😄

[IMPORTANT]
====
The following scripts reboot all of the nodes in the cluster at the same time. This can cause your cluster to be unstable. Another option is to reboot your nodes manually one at a time. Rebooting nodes one-by-one causes considerable downtime in a cluster with many nodes.

> Cluster Operators will not be healthy before rebooting the nodes.

Reviewer: ```suggestion
Cluster Operators will not work correctly before you reboot the nodes.
```
- IBM Style prefers "start" over "boot", but I've retained "boot" here for consistency with the text above this line.
- Avoid giving human characteristics to inanimate objects (e.g., "healthy").

Author: good catch, thanks.
